src/main/java/domain/state/Bonus.java

```diff
@@ -0,0 +1,68 @@
+package domain.state;
+
+import domain.Pins;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static io.OutputResult.SYMBOL_DELIMITER;
+
+public class Bonus implements State {
+
+    private final static int BONUS_FRAME_BOWL_LIMIT = 3;
+    private final static int SECOND_BOWL = 2;
+
+    private List<Pins> pins;
+    private List<State> states;
+
+    public Bonus() {
+        pins = new ArrayList<>();
+        states = new ArrayList<>();
+        states.add(new Ready());
+    }
+
+    @Override
+    public State bowl(Pins downPins) {
+        if (isLastBowl() || isEmptyWhenSecondBowl()) {
```

Comment:
`isClosed()`를 사용하면 어떨까요?

```suggestion
        if (isClosed()) {
```